lib/ransack/nodes/value.rb

```diff
@@ -26,7 +26,7 @@ def cast(type)
         case type
         when :date
           cast_to_date(value)
-        when :datetime, :timestamp, :time, :timestamptz
+        when :datetime, :timestamp, :time, :timestamptz, :datetimeoffset
```

Comment:
@vividmuimui could you please add a comment to indicate `datetimeoffset` is SQLServer only.